odd-platform-api/src/test/java/org/opendatadiscovery/oddplatform/repository/TagRepositoryImplTest.java

```diff
@@ -1,17 +1,217 @@
 class TagRepositoryImplTest extends BaseIntegrationTest {
 
     @Autowired
     private TagRepository tagRepository;
+    @Autowired
+    private DataEntityRepository dataEntityRepository;
+
+    @Test
+    @DisplayName("Creates tag pojo, expecting tag pojo in db")
+    void testCreateTagPojo() {
+        final TagPojo tag = new TagPojo()
+            .setName(UUID.randomUUID().toString())
+            .setImportant(true);
+
+        final TagPojo expectedTagPojo = tagRepository.create(tag);
+
+        assertThat(expectedTagPojo).isNotNull();
+        assertThat(expectedTagPojo.getName()).isNotNull();
+    }
+
+    /**
+     * Test case: Bulk creation of tags
+     * Expected result: 1. All saved tags have ids 2. Saved tags have same names as Input bunch of tags
+     */
+    @Test
+    @DisplayName("Bulk creation of tag pojos, expecting all tags created successfully ")
+    void testBulkCreateTag() {
+        final int numberOfTestTags = 3;
+        final List<TagPojo> testTagsList = createTestTagList(numberOfTestTags);
+        final List<String> testTagsListNames = testTagsList.stream().map(TagPojo::getName).toList();
+
+        final List<TagPojo> actualTagsList = tagRepository.bulkCreate(testTagsList);
+
+        assertThat(actualTagsList).isNotEmpty()
+            .extracting(TagPojo::getId).doesNotContainNull();
+        assertThat(actualTagsList)
+            .extracting(TagPojo::getName)
+            .containsAll(testTagsListNames);
+    }
+
+    /**
+     * Test case: Retrieves all required tags by name
+     * Expected result:
+     * 1.Retrieved list of tags is not empty
+     * 2. All retrieved tags have same names as input list of tags
+     */
+    @Test
+    @DisplayName("Retrieves several tags by names")
+    void testGetTagsByListNames() {
+        final int numberOfTestTags = 3;
+        final List<TagPojo> testTagsList = createTestTagList(numberOfTestTags);
+        final List<TagPojo> expectedTagsList = tagRepository.bulkCreate(testTagsList);
+        final List<String> expectedTagsListNames = expectedTagsList.stream().map(TagPojo::getName).toList();
+
+        final List<TagPojo> actualTagsList = tagRepository.listByNames(expectedTagsListNames);
+
+        assertThat(actualTagsList).isNotEmpty()
+            .extracting(TagPojo::getName).containsAll(expectedTagsListNames);
+    }
+
+    @Test
+    @DisplayName("Creates tags relations with data entity, expecting all tags are connected to data entity")
+    void testCreateRelations() {
+        final int numberOfTestTags = 3;
+        final List<TagPojo> testTagsList = createTestTagList(numberOfTestTags);
+        final List<TagPojo> savedTagsList = tagRepository.bulkCreate(testTagsList);
+        final List<Long> savedTagsListIds = savedTagsList.stream().map(TagPojo::getId).toList();
+        final List<DataEntityPojo> testDataEntityList = dataEntityRepository.bulkCreate(List.of(new DataEntityPojo()));
+        final Long dataEntityId = testDataEntityList.get(0).getId();
```

Comment:
This is not required, but here you can also create tags, which are not linked to data entity, and validate that only linked tags are returned